packages/trackerless-network/src/logic/RandomGraphNode.ts

```diff
@@ -200,13 +190,29 @@ export class RandomGraphNode extends EventEmitter<Events> implements IStreamNode
         if (this.stopped) {
             return
         }
-        this.config.nearbyNodeView.replaceAll(closestNodes.map((descriptor) =>
+        this.updateNearbyNodeView(closestNodes)
+    }
+
+    private updateNearbyNodeView(nodes: PeerDescriptor[]) {
+        this.config.nearbyNodeView.replaceAll(Array.from(nodes).map((descriptor) =>
             new RemoteRandomGraphNode(
                 descriptor,
                 this.config.randomGraphId,
                 toProtoRpcClient(new NetworkRpcClient(this.config.rpcCommunicator.getRpcClientTransport()))
             )
         ))
+        for (const descriptor of this.config.layer1.getKBucketPeers()) {
+            if (this.config.nearbyNodeView.size() < this.config.nodeViewSize) {
```

Comment:
Is this correct? We break out of loop (i.e. don't add the item) if there are are less items than what we want to be in `this.config.nodeViewSize`?